src/hooks/twilio/useVideoRoomState.tsx

```diff
@@ -22,58 +19,57 @@ export const useVideoRoomState = (
   roomName?: string | undefined,
   activeParticipantByDefault = true
 ) => {
-  const [room, setRoom] = useState<Room>();
-  const [videoError, setVideoError] = useState("");
-  const dismissVideoError = useCallback(() => setVideoError(""), []);
-  const [localParticipant, setLocalParticipant] = useState<LocalParticipant>();
-  const [
-    participants,
-    { upsert: upsertParticipant, filter: filterParticipants },
-  ] = useList<ParticipantWithUser<RemoteParticipant>>([]);
-
-  const disconnect = useCallback(() => {
-    setRoom((currentRoom) => {
-      if (currentRoom?.localParticipant?.state !== "connected")
-        return currentRoom;
-
-      currentRoom.localParticipant.tracks.forEach((trackPublication) => {
-        (trackPublication.track as LocalVideoTrack).stop();
-      });
+  const {
+    joinChannel,
+    localParticipant,
+    remoteParticipants,
+    status,
+    disconnect,
+  } = useVideoComms();
 
-      currentRoom.disconnect();
+  const [idToUserMap, setIdToUserMap] = useState<
+    Record<string, WithId<User> | undefined>
```

Comment:
It will be helpful to (re)move `WithId` from the `./src/utils` file, as per the comment above it, if we diminish its direct use i.e. the number of imports

```suggestion
    Record<string, UserWithId | undefined>
```